test/controller/v1/LocalUnitControllerTest.scala

```diff
@@ -0,0 +1,106 @@
+package controller.v1
+
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.client.WireMock
+import com.github.tomakehurst.wiremock.client.WireMock._
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration._
+import com.typesafe.scalalogging.LazyLogging
+import org.scalatest.BeforeAndAfterEach
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.libs.json._
+import play.api.test.Helpers._
+import uk.gov.ons.sbr.models._
+import resources.TestUtils
+
+/**
+ * Created by coolit on 16/04/2018.
+ */
+case class Response(id: String, period: String, unitType: String, parents: Map[String, String], vars: LocalUnit)
+object Response {
+  implicit val unitFormat: OFormat[Response] = Json.format[Response]
+}
+
+class LocalUnitControllerTest extends TestUtils with BeforeAndAfterEach with GuiceOneAppPerSuite with LazyLogging {
+
+  private val version = "v1"
+
+  private val port = 9001
+  private val host = "localhost"
+  private val wireMockServer = new WireMockServer(wireMockConfig().port(port))
+
+  override def beforeEach {
+    wireMockServer.start()
+    WireMock.configureFor(host, port)
+  }
+
+  override def afterEach {
+    wireMockServer.stop()
+  }
+
+  def mockEndpoint(path: String, body: String): Unit = stubFor(get(urlEqualTo(path))
+    .willReturn(
+      aResponse()
+        .withStatus(200)
+        .withHeader("content-type", "application/json")
+        .withBody(body)
+    ))
+
+  "/v1/periods/:date/leus/:id" should {
+    "return a local unit for the given id and period" in {
+      val id = "900000123"
+      val entId = "12345"
+      val period = "201803"
+      val unitType = "LOU"
+
+      val unitLinksJson: JsValue = JsObject(Seq(
+        "id" -> JsString(id),
+        "period" -> JsString(period),
+        "unitType" -> JsString(unitType),
+        "parents" -> JsObject(Map("ENT" -> JsString("12345")))
+      ))
+
+      val louJson: JsValue = JsObject(Seq(
```

Comment:
If any of the optional fields were missing (as they most likely will be), can you still create a LocalUnit?
Where this is important, we typically have one test with all values populated, and one test with only the mandatory values populated.